controllers/grid/queries/form/QueryForm.inc.php

```diff
@@ -323,6 +323,65 @@ function readInputData() {
 		));
 	}
 
+	/**
+	 * @copydoc Form::validate()
+	 */
+	function validate($callHooks = true) {
+		// Display error if anonymity is impacted, i.e. if:
+		// 1) several blind reviewers are selected, or
+		// 2) a blind reviewer and another participant (other than editor) are selected.
+		// Editors are ignored, they can see everything.
+		$query = $this->getQuery();
+		// Queryies only support ASSOC_TYPE_SUBMISSION so far (see above)
+		if ($query->getAssocType() == ASSOC_TYPE_SUBMISSION) {
+			$request = Application::get()->getRequest();
+			$submissionId = $query->getAssocId();
+
+			$reviewAssignmentDao = DAORegistry::getDAO('ReviewAssignmentDAO'); /* @var $reviewAssignmentDao ReviewAssignmentDAO */
+			$stageAssignmentDao = DAORegistry::getDAO('StageAssignmentDAO'); /* @var $stageAssignmentDao StageAssignmentDAO */
+			$userGroupDao = DAORegistry::getDAO('UserGroupDAO'); /* @var $userGroupDao UserGroupDAO */
+
+			// get the selected participants
+			$newParticipantIds = $this->getData('users');
+			$participantsToConsider = $blindReviewerCount = 0;
+			foreach ($newParticipantIds as $participantId) {
+				// get participant roles in this workflow stage
+				$assignedRoles = [];
+				$usersAssignments = $stageAssignmentDao->getBySubmissionAndStageId($submissionId, $query->getStageId(), null, $participantId);
+				while ($usersAssignment = $usersAssignments->next()) {
+					$userGroup = $userGroupDao->getById($usersAssignment->getUserGroupId());
+					$assignedRoles[] = $userGroup->getRoleId();
+				}
+				// get participant review assignemnts
+				$reviewAssignments = $reviewAssignmentDao->getBySubmissionReviewer($submissionId, $participantId);
+				// if participant has no role in this stage and is not a reviewer, display error
+				if (empty($assignedRoles) && empty($reviewAssignments)) {
```

Comment:
Will this prevent a journal manager who is not assigned at all to the submission from creating a discussion?

Reply:
Good catch!!! Thanks a lot!